presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp

```diff
@@ -118,9 +118,12 @@ class PartitionAndSerializeOperator : public Operator {
 
     // Allocate memory.
     auto buffer = dataVector.getBufferWithSpace(totalSize);
+    // getBufferWithSpace() may return a buffer that already has content, so we
+    // only use the space after that.
+    auto rawBuffer = buffer->asMutable<char>() + buffer->size();
+    buffer->setSize(buffer->size() + totalSize);
```

Comment:
@mbasmanova 
Could we add another member getBufferWithSpaceAndSetSize() or include the two lines into getBufferWithSpace()?

Reply:
@spershin We can look into improving the API, but is it not straightforward. 

getBufferWithSpace returns a BufferPtr, not raw pointer. If BufferPtr already has size set, then rawBuffer needs to be computed like so:

```
auto rawBuffer = buffer->asMutable<char>() - totalSize;
```

Alternatively, maybe we can change the API to set size and return raw pointer to the start of the 'totalSize' chunk.

```
auto* rawBuffer = dataVector.getStringBufferWithSpace(totalSize);
```

I can make this change in a separate set of PRs.